pygmt/alias.py

```diff
@@ -195,3 +197,116 @@ def _value(self) -> str | list[str] | None:
             size=self.size,
             ndim=self.ndim,
         )
+
+
+class AliasSystem(UserDict):
+    """
+    Alias system for mapping PyGMT's long-form parameters to GMT's short-form options.
+
+    This class is initialized with keyword arguments, where each key is a GMT option
+    flag, and the corresponding value is an ``Alias`` object or a list of ``Alias``
+    objects.
+
+    This class inherits from ``UserDict``, which allows it to behave like a dictionary
+    and can be passed to the ``build_arg_list`` function. It also provides the ``merge``
+    method to update the alias dictionary with additional keyword arguments.
+
+    Examples
+    --------
+    >>> from pygmt.alias import Alias, AliasSystem
+    >>> from pygmt.helpers import build_arg_list
+    >>>
+    >>> def func(
+    ...     par0, par1=None, par2=None, frame=False, repeat=None, panel=None, **kwargs
+    ... ):
+    ...     aliasdict = AliasSystem(
+    ...         A=[
+    ...             Alias(par1, name="par1"),
+    ...             Alias(par2, name="par2", prefix="+o", separator="/"),
+    ...         ],
+    ...         B=Alias(frame, name="frame"),
+    ...         D=Alias(repeat, name="repeat"),
+    ...         c=Alias(panel, name="panel", separator=","),
+    ...     ).merge(kwargs)
+    ...     return build_arg_list(aliasdict)
+    >>> func(
+    ...     "infile",
+    ...     par1="mytext",
+    ...     par2=(12, 12),
+    ...     frame=True,
+    ...     repeat=[1, 2, 3],
+    ...     panel=(1, 2),
+    ...     J="X10c/10c",
+    ... )
+    ['-Amytext+o12/12', '-B', '-D1', '-D2', '-D3', '-JX10c/10c', '-c1,2']
+    """
+
+    def __init__(self, **kwargs):
+        """
+        Initialize the alias system as a dictionary with current parameter values.
+        """
+        # Store the aliases in a dictionary, to be used in the merge() method.
+        self.aliasdict = kwargs
```

Comment:
Should't this be overriding the UserDict's `data` attribute? https://docs.python.org/3/library/collections.html#collections.UserDict.data

```suggestion
        self.data = kwargs
```

Then I suppose you can use `update` as method name?